src/radical/saga/adaptors/shell/shell_job.py

```diff
@@ -396,7 +397,9 @@ def stage_input(self, shell, jd):
                     source = local
                     target = remote
                     self._logger.info("stage %s to %s" % (source, target))
-                    shell.stage_to_remote(source, target)
+
+                    with self._shell_lock:
```

Comment:
I think it shouldn't be here, `self._shell_lock` is not defined in class `Adaptor`, but in `ShellJobService`

Reply:
Yeah, that was quite stupid - is fixed now, the locks are passed to the calls.